### PR TITLE
Gamelab: SerializedAnimationProps does not need source size

### PR DESCRIPTION
**Pull request by islemaster**
There's no need to specify the `sourceSize` property in the level's animation data (or to serialize it out, for that matter) so this PR moves sourceSize from the `SerializedAnimationProps` type to its ephemeral superset the `AnimationProps` type, and extends the `loadAnimationFromSource` process to read the spritesheet dimensions and populate `sourceSize` at that time.

A brief overview of how animations are loaded:
1. On page load, we receive a `SerializedAnimationList` (a collection of `SerializedAnimationProps` objects) from either the level definition or from the student's saved project.  This contains the animation properties (name, frame size, framerate, etc) but not the spritesheet images themselves.
2. We push those properties into Redux as an `AnimationList` of `AnimationProps` (which are like `SerializedAnimationProps` but they have more fields), and in the process [we call `loadAnimationFromSource`](https://github.com/code-dot-org/code-dot-org/blob/6b24bf68cd49c85f4df69997d4e31e2f8a4062c9/apps/src/gamelab/animationListModule.js#L179) for every animation.
3. `loadAnimationFromSource` makes an asynchronous request for the animation's spritesheet image.  When it's done downloading as a [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) we convert it to a [data URI](https://developer.mozilla.org/en-US/docs/Web/HTTP/data_URIs) and (as of this change) get its dimensions, add a `loadedFromSource` flag, and put all of those on the `AnimationProps` in Redux.

And how they're used:
- Pretty much every place we use the animation now, we are reading its data URI from the `AnimationProps` in Redux - this saves us from making a web request for the spritesheet image again.  That includes loading the animation into p5.play and sending the image to Piskel to load for editing.
- Whenever an edit is made in Piskel we update the corresponding Blob and data URI in Redux so that every place the image is used gets updated.
- When the project gets saved, we upload the image Blob as a PNG image to our animations API, and then we convert the `AnimationList` back into a `SerializedAnimationList` and save that with the project.

### CODE REVIEW

**Comment by islemaster** [comment]
Note: After this is deployed to levelbuilder, I will go update existing gamelab levels to remove sourceSize from their JSON.  This is non-urgent though; the sourceSize field in that JSON will just be ignored after this change.

File: apps/test/unit/gamelab/animationListModuleTest.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import {combineReducers} from 'redux';
+import reducer, {
+    START_LOADING_FROM_SOURCE,
+    DONE_LOADING_FROM_SOURCE
+} from '@cdo/apps/gamelab/animationListModule';
+import {EMPTY_IMAGE} from '@cdo/apps/gamelab/constants';
+import {createStore} from '@cdo/apps/redux';
+import {expect} from '../../util/configuredChai';
+
+// var testUtils = require('../../util/testUtils');
```
Comment (caleybrock) [inline apps/test/unit/gamelab/animationListModuleTest.js]: do we need these if they are commented out? or are they related to the todos?

**Comment by caleybrock** [comment]
👍 

thank you for the explanations!

File: apps/test/unit/gamelab/animationListModuleTest.js (hunk 1 above)
Comment (islemaster) [inline apps/test/unit/gamelab/animationListModuleTest.js]: Oh, nope, just forgot to remove them.  Nice catch.

**Pull request merged**